network/transport/nats/nats_transport.go

```diff
@@ -0,0 +1,30 @@
+package nats
+
+import (
+	"github.com/nats-io/nats.go"
+	"github.com/statechannels/go-nitro/network/transport"
+)
+
+type natsTransport struct {
+	connection transport.Connection
+}
+
+var _ transport.Transport = (*natsTransport)(nil)
+
+func NewNatsTransport(nc *nats.Conn, pubTopicNames []string) *natsTransport {
+	// wouldn't it be better to get messaged directly into the channel?
+	connection := NewNatsConnection(nc, pubTopicNames)
+	natsTransport := &natsTransport{
+		connection: connection,
+	}
+
+	return natsTransport
+}
+
+func (t *natsTransport) PollConnection() (transport.Connection, error) {
```

Comment:
Can you explain why the "Poll" part?

Reply:
Great question. Maybe @calj can elaborate on the thinking behind the naming.